### ISSUE TITLE
Incorrect syntax highlighting with single quotes

### ISSUE THREAD

**Issue by whoami730-R** [scala/vscode-scala-syntax#260]
```scala
val x = ''' + '"'
val y = 1 + 0
```
is highlighted incorrectly and `'''` is assumed to start a string instead of being the string itself whereas
```scala
val x = '"' + '''
val y = 1 + 0
```
is correctly highlighted

**Comment by som-snytt** [comment]
`'''` is bad syntax.
```
3 |  val x = ''' + '"'
  |          ^
  |          empty character literal (use '\'' for single quote)
```

**Comment by whoami730-R** [comment]
@som-snytt intellij seems to not be complaining - Guess I forgot to mention but this is scala 2.12.x

<img width="818" alt="image" src="https://github.com/scala/vscode-scala-syntax/assets/107390598/0b04f9c3-1754-488a-bfc3-bcf49d1e9b38">

**Comment by whoami730-R** [comment]
https://scastie.scala-lang.org/ubVQfrgwSgyB3LWIa0x77A also seems to not complain

**Comment by som-snytt** [comment]
yes, it was a bug in 2.12. I remember the fix was deferred to a major release.

Actually github says 2.12.3. Oh wait, "Enforcement is under `-Xsource:2.13`." It was deprecated at 2.12.3, errors at 2.13.

https://scastie.scala-lang.org/2NAOmTP4SZeZHft2rRlYvQ

**Comment by som-snytt** [comment]
The caution is that people accustomed to other languages have various assumptions about quote marks, such as that they introduce strings.

If you allow this syntax, then you must explain that `'''` is not analogous to `"""`.

Since the need is rare, it's not burdensome to write `'\''`. When constructing a string, prefer `s"so-called 'fun'!"`.
